### ISSUE TITLE
Implement a red border around invalid link formatting

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#11793]
**Description**
This issue is split from #1838 and the PR #11286.

It's blocked by #11484

When a user inserts a link that's deemed to be invalid, the classic editor displays a red dashed border around the formatting to indicate a potential issue:
![screen shot 2018-11-05 at 4 53 14 pm](https://user-images.githubusercontent.com/677833/47987855-90880500-e11b-11e8-92b5-dabbbc10e6e4.png)

This formatting isn't present in Gutenberg (or in the classic block) but is an important part of link validation.

Currently the Format API/Rich Text implementation doesn't support a way to indicate errors in formatting on the editor side only, this is tracked in #11484. Once that's solved, the red border should be implemented in Gutenberg.

**Comment by afercia** [comment]
I'd kindly ask to set on this issue the same milestone that was set on #1838 as the original issue isn't fully solved and color alone is not sufficient as feedback. Thank you.

**Comment by talldan** [comment]
I've added it to the milestone, but I'd temper expectations—given this is blocked it might not be resolved in time for this to be added.

**Comment by jordesign** [comment]
Just checking through older issues - and given this hasn't been revisited in more than 4 years - do we consider this still relevant and important (to have exact parity with the Classic Editor)?

**Comment by afercia** [comment]
Following the instructions from [the original PR](https://github.com/WordPress/gutenberg/pull/11286):

```
Try inserting a variety of invalid links (e.g. http:// test.com/, htt#p://test.com)
```

I can't seem to receive any visual warning the link format is invalid. Not even sure the validation is working as expected. Maybe I'm missing something.

Anyways, color only is not enough to signal an invalid link format. There should be some visible text or a visual indication that isn't based on color alone. Visible text would be preferred.

Screenshots from current trunk:


`htt#p://test.com`

<img width="404" alt="Screenshot 2023-08-21 at 11 36 46" src="https://github.com/WordPress/gutenberg/assets/1682452/baa67224-4e60-47ff-a1e2-cb1279d81294">


`http:// test.com/`

<img width="416" alt="Screenshot 2023-08-21 at 11 37 38" src="https://github.com/WordPress/gutenberg/assets/1682452/d3c946fb-3d69-4432-bf00-56755854b72e">

**Comment by t-hamano** [comment]
This issue is till exists.

Should the visual text `only` be displayed after a possibly incorrect URL is entered? Or should it always be displayed?

If we follow the Classic Editor specs, it would be the former.
